src/test/scala/com/datamountaineer/streamreactor/connect/schemas/TestConverterUtil.scala

```diff
@@ -233,5 +233,262 @@ class TestConverterUtil extends TestUtilsBase with ConverterUtil {
+      val originalRecord = sinkRecordWithKeyHeaders()
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map("*" -> "*"),
+        headerFields = Map.empty
+      )
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe originalRecord.valueSchema().fields().size
+
+      simpleJsonConverter.fromConnectData(combinedRecord.valueSchema(), combinedRecord.value()).toString shouldBe(
+        simpleJsonConverter.fromConnectData(originalRecord.keySchema(), originalRecord.key()).toString
+        )
+    }
+
+    "select * from headers fields for struct" in {
+
+      val originalRecord = sinkRecordWithKeyHeaders()
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map.empty,
+        headerFields = Map("*" -> "*")
+      )
+
+      val expected = "{\"header_field_1\":\"foo\",\"header_field_2\":\"bar\",\"header_field_3\":\"boo\"}"
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe originalRecord.headers().size
+
+      simpleJsonConverter.fromConnectData(combinedRecord.valueSchema(), combinedRecord.value()).toString shouldBe expected
+    }
+
+    "should return empty record for struct" in {
+
+      val originalRecord = sinkRecordWithKeyHeaders()
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map.empty,
+        headerFields = Map.empty
+      )
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe 0
+    }
+
+    "should retain key and headers for struct" in {
+
+      val originalRecord = sinkRecordWithKeyHeaders()
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map.empty,
+        headerFields = Map.empty,
+        retainKey = true,
+        retainHeaders = true
+      )
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe 0
+      combinedRecord.keySchema() shouldBe originalRecord.keySchema()
+      combinedRecord.key() shouldBe originalRecord.key()
+      combinedRecord.headers() shouldBe originalRecord.headers()
+    }
+
+    "should ignore fields for struct" in {
+      val originalRecord = getTestRecord
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map("*" -> "*"),
+        ignoreFields = Set("int_field"),
+        keyFields = Map.empty,
+        headerFields = Map.empty,
+        retainKey = true,
+        retainHeaders = true
+      )
+
+      val expected = "{\"id\":\"sink_test-1-1\",\"long_field\":1,\"string_field\":\"foo\"}"
+
+      simpleJsonConverter.fromConnectData(combinedRecord.valueSchema(), combinedRecord.value()).toString shouldBe expected
+    }
+
+    "should add fields, key fields and headers for schemaless JSON" in {
```

Comment:
We would require a test with a nested field as well. The nested field will be also util.Map